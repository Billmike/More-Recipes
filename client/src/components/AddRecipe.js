import React from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

const AddRecipe = (props) => {
    console.log(props)
	return (
	   <div>
       <h1 className="container">Add Recipe</h1>
        <RecipesForm onSubmit={(recipe) => {
            props.dispatch(addRecipe(recipe));
            props.history.push('/')
        }} />
       </div>	
	)
}

export default connect()(AddRecipe);
