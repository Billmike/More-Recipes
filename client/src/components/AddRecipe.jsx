import React, { Component } from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

export class AddRecipe extends Component {
  onSubmit = (recipe) => {
    this.props.addRecipe(recipe);
    this.props.history.push('/dashboard');
  };

  render() {
    return (
      <div>
        <h1 className="container"> Add Recipe </h1>
        <RecipesForm
          onSubmit={this.onSubmit}
        />
      </div>
      );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addRecipe: (recipe) => dispatch(addRecipe(recipe))
  }
}

// const AddRecipe = (props) => {
//     console.log(props)
// 	return (
// 	   <div>
//        <h1 className="container">Add Recipe</h1>
//         <RecipesForm onSubmit={(recipe) => {
//             props.dispatch(addRecipe(recipe));
//             props.history.push('/')
//         }} />
//        </div>	
// 	)
// }

export default connect(undefined, mapDispatchToProps)(AddRecipe);
