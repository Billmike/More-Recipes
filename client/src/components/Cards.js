import React, { Component } from 'react';

class Cards extends Component {
	render () {
		return (
			<div>
			  <div className="card-group">
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src="../src/assets/img/pizzza.jpg" alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true"></i></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true"></i></a> 
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true"></i></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src="../src/assets/img/pizzza.jpg" alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">180 <i className="fa fa-eye" aria-hidden="true"></i></a>
              <a href="#" className="border border-secondary rounded btn" data-toggle="modal" data-target="#exampleModal">  150 <i className="fa fa-heart" aria-hidden="true"></i></a> 
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 70 <i className="fa fa-comments" aria-hidden="true"></i></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src="../src/assets/img/pizzza.jpg" alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action. </p>
              <a href="./template/recipe_detail.html" className="border btn border-secondary rounded">67 <i className="fa fa-eye" aria-hidden="true"></i></a>
              <a href="#" className="border btn border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  49 <i className="fa fa-heart" aria-hidden="true"></i></a> 
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 90 <i className="fa fa-comments" aria-hidden="true"></i></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
        </div>
		</div>
		)
	}
}

export default Cards;
