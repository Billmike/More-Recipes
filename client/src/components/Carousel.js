import React, { Component } from 'react';

class Carousel extends Component {
	render () {
		return (
			<div>
			  <div id="carouselExampleIndicators" className="carousel slide" data-ride="carousel">
          <ol className="carousel-indicators">
            <li data-target="#carouselExampleIndicators" data-slide-to="0" className="active"></li>
            <li data-target="#carouselExampleIndicators" data-slide-to="1"></li>
            <li data-target="#carouselExampleIndicators" data-slide-to="2"></li>
          </ol>
          <div className="carousel-inner">
            <div className="carousel-item active">
              <img className="d-block w-100 sizing" src="../src/assets/img/pancakes.jpeg" alt="First slide" />
              <div className="carousel-caption d-none d-md-block">
                  <h3 className="carousel-h3">Handcrafted recipes made by local chefs</h3>
                  <p className="carousel-p">... the little things matter</p>
                </div>
            </div>
            <div className="carousel-item">
              <img className="d-block w-100 sizing" src="../src/assets/img/cookies.jpeg" alt="Second slide" />
              <div className="carousel-caption d-none d-md-block">
                  <h3 className="carousel-h3-light">Tasty recipes to spice up your dinner table.</h3>
                  <p className="carousel-p-light">... you really can't resist these delights.</p>
                </div>
            </div>
            <div className="carousel-item">
              <img className="d-block w-100 sizing" src="../src/assets/img/dessert.jpeg" alt="Third slide" />
              <div className="carousel-caption d-none d-md-block">
                  <h3 className="carousel-h3">Make every meal memorable.</h3>
                  <p className="carousel-p">... no more boring nights at home.</p>
                </div>
            </div>
          </div>
          <a className="carousel-control-prev" href="#carouselExampleIndicators" role="button" data-slide="prev">
            <span className="carousel-control-prev-icon" aria-hidden="true"></span>
            <span className="sr-only">Previous</span>
          </a>
          <a className="carousel-control-next" href="#carouselExampleIndicators" role="button" data-slide="next">
            <span className="carousel-control-next-icon" aria-hidden="true"></span>
            <span className="sr-only">Next</span>
          </a>
        </div>
			</div>
		)
	}
}

export default Carousel;
