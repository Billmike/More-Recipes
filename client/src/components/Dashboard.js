import React from 'react';
import { Link } from 'react-router-dom';
import pizza from '../assets/img/pizzza.jpg';
import '../assets/css/style.css';

const Dashboard = () => {
    return(
        <div className="container-fluid">
          <div className="row">
            <main className="container">
              <h1>My Recipes</h1>
        <div className="card-group">
          <div className="card">
            <a href="./recipe_detail.html"><img className="card-img-top" src={pizza} alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.</p>
              <a href="edit.html" className="btn btn-outline-success" role="button">Edit <i className="fa fa-pencil-square-o"></i></a> 
              <button type="button" className="btn btn-outline-danger" data-toggle="modal" data-target="#exampleModal">
                  Delete <i className="fa fa-trash-o" aria-hidden="true"></i>
                </button>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
          <div className="card">
            <a href="./recipe_detail.html"><img className="card-img-top" src={pizza} alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.</p>
              <a href="edit.html" className="btn btn-outline-success" role="button">Edit <i className="fa fa-pencil-square-o"></i></a> 
              <button type="button" className="btn btn-outline-danger" data-toggle="modal" data-target="#exampleModal">
                  Delete <i className="fa fa-trash-o" aria-hidden="true"></i>
                </button>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
          <div className="card">
            <a href="./recipe_detail.html"><img className="card-img-top" src={pizza} alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.</p>
              <a href="edit.html" className="btn btn-outline-success" role="button">Edit <i className="fa fa-pencil-square-o"></i></a> 
              <button type="button" className="btn btn-outline-danger" data-toggle="modal" data-target="#exampleModal">
                Delete <i className="fa fa-trash-o"></i>
              </button>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
        </div>
            </main>
          </div>
        </div>
    )
}

export default Dashboard;
