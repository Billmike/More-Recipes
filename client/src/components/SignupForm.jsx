import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import InputFieldGroup from './InputFieldGroup';
import validateInput from '../../../server/validators/validatesignup';
import '../assets/css/signup.css';

class SignupForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      errors: {},
      isLoading: false,
    };
    this.onUsernameChange = this.onUsernameChange.bind(this);
    this.onPasswordChange = this.onPasswordChange.bind(this);
    this.onEmailChange = this.onEmailChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onUsernameChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  onEmailChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  onPasswordChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }


  onSubmit(event) {
    event.preventDefault();

    if (this.isValid()) {
      this.setState({ errors: {}, isLoading: true });
      this.props
        .signupRequest(this.state)
        .then((response) => {
          console.log(response);
        })
        .catch((errors) => {
          console.log(errors.response);
          this.setState({
            errors: errors.response.data,
            isLoading: false,
          });
        });
    }
  }

  isValid() {
    const { errors, valid } = validateInput(this.state);

    if (!valid) {
      this.setState({ errors });
    }

    return valid;
  }

  render() {
    const { errors } = this.state;
    return (
      <form onSubmit={this.onSubmit}>
        <div id="login-box">
          <div className="left">
            <h1>Sign up</h1>

            <InputFieldGroup
              errors={errors.username}
              placeholder="Username"
              onChange={this.onUsernameChange}
              value={this.state.username}
              field="username"
            />

            <InputFieldGroup
              errors={errors.email}
              placeholder="Email Address"
              onChange={this.onEmailChange}
              value={this.state.email}
              field="email"
            />

            <InputFieldGroup
              errors={errors.password}
              placeholder="Password"
              onChange={this.onPasswordChange}
              value={this.state.password}
              field="password"
            />
            <input
              disabled={this.state.isLoading}
              type="submit"
              name="signup_submit"
              value="Sign up"
            />
          </div>

          <div className="right">
            <span className="loginwith">
              Sign in with<br />social network
            </span>

            <button className="social-signin facebook">
              Log in with facebook
            </button>
            <button className="social-signin twitter">
              Log in with Twitter
            </button>
            <button className="social-signin google">
              Log in with Google+
            </button>
          </div>
          <div className="or">OR</div>
        </div>
      </form>
    );
  }
}

SignupForm.propTypes = {
  signupRequest: PropTypes.func.isRequired,
};

export default SignupForm;
