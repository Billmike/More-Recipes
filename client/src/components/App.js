import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import Favorite from './Favorite';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import NotFoundPage from './NotFoundPage';

class App extends Component {
    render() {
        return(
            <div>
                <BrowserRouter>
                    <div>
                        <Navbar />
                        <Switch>
                        <Route exact path="/" component={Homepage} />
                        <Route path="/dashboard" component={Dashboard} />
                        <Route path="/favorite" component={Favorite} />
                        <Route path="/add" component={AddRecipe} />
                        <Route path="/edit" component={EditRecipe} />
                        <Route path="/profile" component={ProfilePage} />
                        <Route component={NotFoundPage} />
                    </Switch>
                    </div>
                </BrowserRouter>
            </div>
        )
    }
}

export default App;
