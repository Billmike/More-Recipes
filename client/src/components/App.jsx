import React from 'react';
import { BrowserRouter, browserHistory, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import RecipeDetails from './RecipeDetails';
import Favorite from './Favorites';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import LoginPage from './LoginPage';
import RegistrationPage from './RegistrationPage';
import RequireAuth from '../utils/RequireAuth';
import NotFoundPage from './NotFoundPage';

const App = () => (
  <div>
    <BrowserRouter history={browserHistory}>
      <div>
        <Navbar />
        <Switch>
          <Route exact path="/" component={Homepage} />
          <Route path="/dashboard" component={RequireAuth(Dashboard)} />
          <Route path="/favorites" component={RequireAuth(Favorite)} />
          <Route path="/add" component={RequireAuth(AddRecipe)} />
          <Route path="/edit/:id" component={RequireAuth(EditRecipe)} />
          <Route path="/recipe/:id" component={RecipeDetails} />
          <Route path="/profile" component={RequireAuth(ProfilePage)} />
          <Route path="/login" component={LoginPage} />
          <Route path="/register" component={RegistrationPage} />
          <Route component={NotFoundPage} />
        </Switch>
      </div>
    </BrowserRouter>
  </div>
);

export default App;
