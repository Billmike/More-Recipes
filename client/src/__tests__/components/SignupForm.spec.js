import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { SignupForm } from '../../components/SignupForm';

test('Should render Signup Form correctly', () => {
  const wrapper = shallow(<SignupForm />);
  expect(wrapper).toMatchSnapshot();
});

