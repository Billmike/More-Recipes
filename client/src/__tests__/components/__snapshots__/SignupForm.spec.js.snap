// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupForm component /> Should render the sign-up form correctly 1`] = `
<div
  className="container #signupForm"
>
  <div
    className="card text-center card-form has-feedback"
  >
    <div
      className="card-body"
    >
      <form
        id="submitSignupForm"
        onSubmit={[Function]}
      >
        <h1
          className="sign-up-h3"
        >
          Register
        </h1>
        <div
          className="form-group"
        >
          <input
            autoComplete="username"
            className="form-control form-control-lg"
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            autoComplete="email"
            className="form-control form-control-lg"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            type="email"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            autoComplete="new-password"
            className="form-control form-control-lg"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <input
          className="btn submit-btn size-bt btn-block"
          id="submitButton"
          type="submit"
          value="Submit"
        />
      </form>
      <p>
         Have an account?
        <Link
          className="account"
          replace={false}
          to="/login"
        >
           
            Sign in
        </Link>
      </p>
    </div>
  </div>
</div>
`;
